### PR TITLE
User set retrieve availabilities optional

### LINKED ISSUE #225: DB user updates
* Add email address
* only retrieve availabilities if requested

### PR DESCRIPTION
**Pull request by Plouc314**
* Do not retrieve user availabilities in `UserTable.get` anymore
* Make `UserTable.retrieveAvailabilities` public
* Update tests
* Retrieve assigned flights in calendar view model

Close #225

### CODE REVIEW

File: app/src/main/java/ch/epfl/skysync/viewmodel/CalendarViewModel.kt [hunk 1]
```diff
@@ -31,31 +29,32 @@ data class CalendarUiState(
  * ViewModel for the Calendar screen
  *
  * @param uid The Firebase authentication uid of the user
- * @param userTable The user table
- * @param availabilityTable The availability table
+ * @param repository App repository
  */
 class CalendarViewModel(
     private val uid: String,
```
Comment (Prolas) [inline app/src/main/java/ch/epfl/skysync/viewmodel/CalendarViewModel.kt:35]: Is there an advantage to passing the user id rather than the object User ?

Comment (Plouc314) [reply app/src/main/java/ch/epfl/skysync/viewmodel/CalendarViewModel.kt:35]: That will be possible once #263 is done

Comment (Prolas) [reply app/src/main/java/ch/epfl/skysync/viewmodel/CalendarViewModel.kt:35]: Perfect

**Review by Prolas — APPROVED** [review]
LGTM

File: app/src/androidTest/java/ch/epfl/skysync/viewmodel/CalendarViewModelTest.kt [hunk 2]
```diff
@@ -32,8 +33,7 @@ class CalendarViewModelTest {
     dbs.clearDatabase(db)
     dbs.fillDatabase(db)
     composeTestRule.setContent {
-      calendarViewModel =
-          CalendarViewModel.createViewModel(dbs.admin1.id, userTable, availabilityTable)
+      calendarViewModel = CalendarViewModel.createViewModel(dbs.admin1.id, Repository(db))
       val uiState = calendarViewModel.uiState.collectAsStateWithLifecycle()
       Text(text = uiState.value.user?.firstname ?: "Bob")
```
Comment (Prolas) [inline app/src/androidTest/java/ch/epfl/skysync/viewmodel/CalendarViewModelTest.kt:38]: It seems the user is always null

**Pull request merged**